src/SDK/Resource/ResourceInfoFactory.php

```diff
@@ -13,17 +13,17 @@
 class ResourceInfoFactory
 {
     /**
-     * Merges resources into a new one.
+     * Merges resources into a new one. Duplicate keys from later resources take priority and will overwrite
+     * values from earlier.
      *
-     * @param ResourceInfo ...$resources
-     * @return ResourceInfo
+     * @see https://github.com/open-telemetry/opentelemetry-specification/blob/v1.20.0/specification/resource/sdk.md#merge
      */
     public static function merge(ResourceInfo ...$resources): ResourceInfo
     {
         $attributes = [];
 
         foreach ($resources as $resource) {
-            $attributes += $resource->getAttributes()->toArray();
+            $attributes = array_merge($attributes, $resource->getAttributes()->toArray());
```

Comment:
Can't use `array_merge`, the attributes array can contain numeric keys.

Reply:
> the spec doesn't dictate the argument order

I think it does..."old" and "updating" resource params are listed in that order:

> Required parameters:
> * the old resource
> * the updating resource whose attributes take precedence

https://github.com/open-telemetry/opentelemetry-specification/blob/v1.20.0/specification/resource/sdk.md#merge

And in the case that more than two Resources are provided, I think it then follows that later ones take precedence.

> attributes array can contain numeric keys

https://github.com/open-telemetry/opentelemetry-specification/blob/v1.20.0/specification/common/README.md#attribute says `The attribute key MUST be a non-null and non-empty string` - perhaps we need to cast numeric keys to strings, which should resolve not being able to use `array_merge`... It's certainly not covered by any tests, and probably should be. What do you think?

Reply:
@Nevay you were right about `array_merge` (I didn't realise PHP forces numeric-looking keys to numbers internally), so I've instead applied array_reverse to deal with precedence. I added some tests around numeric keys + merging.

Reply:
Does the spec specify that the parameter order must/should be followed? I always assumed that this is entirely up to the implementation.
(fwiw I'm using `Resource::merge(...$this->resources, _default: Resource::getDefault())` in my implementation which can't be written as concisely with ascending priority.)

Reply:
Not exactly, but it does have this `MUST`:

> `If a key exists on both the old and updating resource, the value of the updating resource MUST be picked (even if the updated value is empty)`

To me, that's saying that if you start with a resource, then merge a second resource, then the second resource's keys take precedence and should overwrite the first.

Looking at other language implementations, `merge` is usually a method on a resource eg `$resourceOne->merge($resourceTwo)` which I think is clearer as to which is the old and the updating resource, and the spec was probably written with that in mind.